eos-paygd/main.c

```diff
@@ -240,6 +295,9 @@ main (int   argc,
   GOptionContext *context;
   gboolean enforcing_mode = TRUE;
 
+  log_io = g_io_channel_new_file (LOG_FILE_NAME, "a", NULL);
+  g_log_set_writer_func (log_writer, log_io, NULL);
```

Comment:
Hum, I thought this would try to free `user_data` after each call to the callback. Are you sure the way you describe is how it works?

I supposed I could also close the io channel on the last line before returning from main as well.